ФЗ12

y8789727 opened the pull request:
Выполнено согласно постановке

y8789727 committed:
ФЗ12

Выполнено согласно постановке

y8789727 committed:
ФЗ12

Исправления стиля

y8789727 committed:
ФЗ12

Исправления стиля

y8789727 committed:
ФЗ12

Исправлены автотесты под новые требования

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/dal/FilmRepository.java:
```diff
@@ -0,0 +1,99 @@
+            "WHERE f.film_id = ? " +
+            "ORDER BY f.film_id";
+
+//    private static final String INSERT_FILM_QUERY =
```
Не оставляем закомментированного кода в проекте.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/dal/GenreRepository.java:
```diff
@@ -0,0 +1,58 @@
+
+    public void saveFilmGenres(Film f, Set<Genre> genres) {
+        delete(DELETE_FILM_GENRES_QUERY, f.getId());
+        genres.forEach(g -> insert(INSERT_FILM_GENRES_QUERY, f.getId(), g.getId()));
```
Это нехорошее решение делать запросы в БД в цикле. Мы грузим базу, сеть. Нужно стараться делать минимальное количество запросов.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/dal/UserRepository.java:
```diff
@@ -0,0 +1,124 @@
+            "WHERE fr.user_id = ? " +
+            "ORDER BY fr.friend_id";
+
+//    private static final String INSERT_USER_QUERY =
```
Не оставляем так

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/dal/UserRepository.java:
```diff
@@ -0,0 +1,124 @@
+            "       JOIN Friends f1\n" +
+            "         ON f1.USER_ID = ?\n" +
+            "        AND f1.FRIEND_ID = u.USER_ID\n" +
+//            "        AND f1.CONFIRMED\n" +
```
Проверьте везде по проекту.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -71,6 +72,22 @@ private void validateFilm(Film film) {
+
+        if (!film.getGenres().isEmpty()) {
+            film.getGenres().forEach(g -> {
+                    Optional<Genre> genreOpt = filmStorage.getGenreById(g.getId());
```
Запросы к БД в цикле.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java:
```diff
@@ -0,0 +1,98 @@
+
+    @Override
+    public Collection<Film> getAll() {
+        return filmRepository.findAll().stream().map(this::fillReferences).toList();
```
Запросы к БД в цикле.

y8789727 committed:
ФЗ12

Изменения по замечаниям ревью
- убран закомментированный код
- исключены запросы к БД в цикле
